[Joelwang22] iP

Joelwang22 opened the pull request:
# DukeLite

> Making things easy is hard - Ted Nelson

So let DukeLite do it for you! It makes keeping track of tasks really easy!

DukeLite is:
- text-based
- **VERY** easy to learn
- ~~FAST~~ **_SUPER_** FAST to use

All you need to do is:
1. Download it from [here](https://github.com/Joelwang22/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is absolutely **FREE**!

Features Roadmap:
- [x] Managing tasks
- [x] Variety of tasks (Events, Deadlines, Todo)
- [ ] Reminders
- [ ] GUI

If you love java, use DukeLite as practice. Here's the `main` method:


```Ruby
public static void main(String[] args){
    new Duke("./data/saveData.txt").start();
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

fubsywubsy committed:
Level-0

fubsywubsy committed:
level-1

fubsywubsy committed:
level-1

fubsywubsy committed:
level-2

fubsywubsy committed:
level-3

fubsywubsy committed:
level-4

fubsywubsy committed:
level-5 error handling added

fubsywubsy committed:
level-5

fubsywubsy committed:
level-6

fubsywubsy committed:
A-TextUiTesting

fubsywubsy committed:
A-TextUiTesting

fubsywubsy committed:
A-TextUiTesting

fubsywubsy committed:
Level-7

fubsywubsy committed:
extracted task classes out of duke

fubsywubsy committed:
extracted TaskException class out of duke

fubsywubsy committed:
branch-Level-8

fubsywubsy committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/TaskException.java
#	src/main/java/Todo.java

fubsywubsy committed:
Merge branch 'branch-Level-8'

fubsywubsy committed:
bugfix

fubsywubsy committed:
A-MoreOOP

fubsywubsy committed:
Merge branch 'add-gradle-support'

fubsywubsy committed:
A-Packages

fubsywubsy committed:
A-Gradle

fubsywubsy committed:
A-JUnit

fubsywubsy committed:
Level-9

fubsywubsy committed:
A-JavaDoc

fubsywubsy committed:
A-CodingStandard

fubsywubsy committed:
Merge tag 'A-JavaDoc'

# Conflicts:
#	src/main/java/duke/util/Parser.java

fubsywubsy committed:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/util/Ui.java

fubsywubsy committed:
Level-10

Zack-Tay commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,46 @@
+import duke.util.Parser;
+import duke.util.Ui;
+
+import java.util.*;
```
I think you should avoid wildcard imports and instead list out all the imports that you require.

Zack-Tay commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,46 @@
+    private Ui ui;
+    private Parser parser;
+
+    public Duke(String filePath) {
```
I like your OOP abstraction ideas which makes your main class look clean and easy to understand. LGTM!

Zack-Tay commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,103 @@
+            return this.getType() + "[X] " + this.getName() + " " + this.getTime();
+        } else {
+            return this.getType() + "[ ] " + this.getName() + " " + this.getTime();
+        }
```
I believe this can be shortened by making use of the isDone() method in between the [ ] brackets, you might want to look into that

Zack-Tay commented on src/main/java/duke/util/TimeManager.java:
```diff
@@ -0,0 +1,47 @@
+     *         "dd MMMM yyyy, h:mm a" if the input matches any of the known formats;
+     *         otherwise, returns the original input string.
+     */
+    public static String parseTime(String input) {
```
I think it would be a nice touch to also be able to include dates but without the time for users.

Zack-Tay commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,96 @@
+public class Ui {
+    private final String LINE = "    ____________________________________________________________";
+    private final String INDENT = "     ";
+    private final String NAME = "Alfred";
```
your chatbot name is Alfred so ideally your main class and packages name should be changed to Alfred as well.

Zack-Tay reviewed (commented):
Good use of OOP to abstraction logic away from the main class and make the code more extendable in the future! There are some changes that you might want to consider as I have listed.

Zack-Tay reviewed (commented):
Overall, LGTM, Just a few nits to fix, and don't forget to start to add javaDocs to make your workload lesser in the coming weeks!

fubsywubsy committed:
added scoll functionality to chats

fubsywubsy committed:
Merge tag 'Level-10'

fubsywubsy committed:
Merge branch 'master' into Level-10

fubsywubsy committed:
added circlular chatter images

justincred commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+package duke;
```
Think it might be nicer to think of a name for the chatbot, I saw that the prof said somewhere that should have been done in level-0 but don't know if will penalise

justincred commented on src/main/java/duke/task/TaskConverter.java:
```diff
@@ -0,0 +1,92 @@
+ * This class supports saving tasks to and loading tasks from a string-based format, facilitating
+ * persistence in text files or similar storage mechanisms.
+ */
+public class TaskConverter {
```
Interesting class to make it clearer for converting task objects to strings which I think is a great idea!

justincred commented on src/main/java/duke/task/TaskConverter.java:
```diff
@@ -0,0 +1,92 @@
+     * @param list The array of strings to search through.
+     * @return The index of the string if found, or -1 if not found.
+     */
+    public static int finder(String checker, String[] list) {
```
might think of renaming checker to something more intuitive such as stringToFind or something similar!

justincred commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,102 @@
+     *
+     * @return An ArrayList of {@code Task} objects representing the current tasks.
+     */
+    public ArrayList<Task> giveList(){
```
might want to consider the more usual way of naming this such as getList

justincred commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,102 @@
+     *
+     * @return The size of the task list.
+     */
+    public int size(){
```
similar to the above, maybe rename it to getSize?

justincred commented on src/main/java/duke/test/ParserTest.java:
```diff
@@ -0,0 +1,40 @@
+package duke.test;
```
hi i think your test is in the wrong directory it should be parallel to main like this 
![image](https://github.com/nus-cs2103-AY2324S2/ip/assets/95204474/d7a63728-91a1-4d80-9dc1-6a4b2536ec09)

justincred reviewed (commented):
Overall nicely done with some interesting classes for nicer OOP with only some minor name nitpicks

fubsywubsy committed:
Add assertions check for changes in taskList size after
deletion of tasks
Add assertiions for null checking during addition of task
Add assertions to check if getTask index is within taskList

fubsywubsy committed:
Improve code quality
Made happy path more prominent
Extract methods

fubsywubsy committed:
Add streams to showList()

Joelwang22 committed:
Merge pull request #1 from Joelwang22/branch-A-Assertions

Add assertions check for changes in taskList size after

fubsywubsy committed:
Merge branch 'master' into branch-A-Streams

Joelwang22 committed:
Merge branch 'master' into branch-A-CodeQuality

Joelwang22 committed:
Merge pull request #2 from Joelwang22/branch-A-CodeQuality

Improve code quality

fubsywubsy committed:
Merge remote-tracking branch 'origin/master' into branch-A-Streams

fubsywubsy committed:
Add functionality for snoozing, postponing and rescheduling
Still buggy

fubsywubsy committed:
Add snooze, reschedule, postpone functionality

fubsywubsy committed:
Revert "Add snooze, reschedule, postpone functionality"

This reverts commit 7bb024bafcc0d2ba17e0dfd89f86e9a883fe8d3b.

fubsywubsy committed:
Revert "Revert "Add snooze, reschedule, postpone functionality""

This reverts commit 47bf9f9fc8037cca4a3cb4006864674695b72b7f.

fubsywubsy committed:
made ui show some errors instead of throwing exceptions

fubsywubsy committed:
The final release of alfred

fubsywubsy committed:
renamed ui image

fubsywubsy committed:
fixed user guide
